### PR TITLE
Added transformation config to support automl efficientdet models

### PR DESCRIPTION
**Pull request by lazarevevgeny**
Description: More generic solution based on https://github.com/openvinotoolkit/openvino/pull/2691. No need to perform constant folding before providing model as input.

Conversion string is the following
```
./mo.py --input_model efficientdet-d4_frozen.pb --transformations extensions/front/tf/automl_efficientdet.json --input_shape "[1,1024,1024,3]"
```

The generated IR contains mul/subtract operations but does not include image resize and padding. These steps must be done manually in the application before providing model as input.

JIRA: 38927

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR: the original model is not reshape-able over spatial/batch dimensions
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests: Done for applicable changes
* [x]  Framework operation tests: N/A no new operation enabled
* [x]  Transformation tests: N/A not applicable for such cases
* [x]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py: 38934
* [x]  Model Optimizer IR Reader check: Done

Documentation:
* [x]  Supported frameworks operations list: N/A No new operations enabled
* [x]  Supported **public** models list: N/A 
* [x]  New operations specification: N/A no new operation enabled
* [x]  Guide on how to convert the **public** model: Done
* [x]  User guide update: Not applicable

### CODE REVIEW

**Comment by lazarevevgeny** [comment]
@dkurt , do you know what should be changed in the json file so the conversion work properly for other versions of the model (d0, d1...)?

**Comment by lazarevevgeny** [comment]
> @dkurt , do you know what should be changed in the json file so the conversion work properly for other versions of the model (d0, d1...)?

Found that the only difference is in the input image shape which can be found from here https://github.com/google/automl/blob/96e1feef7a39bd343ce19326b134d62fb640a356/efficientdet/hparams_config.py#L304

**Comment by dkurt** [comment]
> --output "concat,concat_1"

why do we need to specify it?

**Comment by lazarevevgeny** [comment]
> > --output "concat,concat_1"
> 
> why do we need to specify it?

To get rid of the post-processing part of the model and just insert DetectionOutput.

**Comment by dkurt** [comment]
> To get rid of the post-processing part of the model and just insert DetectionOutput.

But it's not used during the transformation - all the unconnected nodes are removed anyway. So original PR can work without it (and I think this version as well)

**Comment by dkurt** [comment]
@lazarevevgeny maybe it's better to just remove all the unconnected nodes?

File: model-optimizer/extensions/front/Pack.py [hunk 1]
```diff
@@ -15,25 +15,26 @@
...
-            expand_dims_node.out_port(0).connect(out_node.in_port(ind))
-        # Replace edge from out port 0 of the matched node with a edge from node out_node.id with port 0.
-        # The "explicit" version of the return value is: [(out_node.id, 0)])
+            unsqueeze_node = create_op_with_const_inputs(graph, Unsqueeze, {1: int64_array([node.axis])},
+                                                         {'name': node.soft_get('name', node.id) + '/Unsqueeze'})
+            node.in_port(ind).get_connection().set_destination(unsqueeze_node.in_port(0))
+            unsqueeze_node.out_port(0).connect(out_node.in_port(ind))
+
+        rename_nodes([(node, pack_name + '/TBR'), (out_node, pack_name)])
         return [out_node.id]
```
Comment (dkurt) [inline model-optimizer/extensions/front/Pack.py:39]: Have you faced Swish layer fusion bug? Does it work correctly with this version?

https://github.com/openvinotoolkit/openvino/pull/2691/files#diff-b1c9357fb1b7f85fdea7959bf6184e5b29a8ec5ff9fc509a327bb60de804411e

**Comment by lazarevevgeny** [comment]
> @lazarevevgeny maybe it's better to just remove all the unconnected nodes?

All unconnected nodes will be removed automatically by the MO. Can we talk in the work chat? :)

File: model-optimizer/extensions/front/Pack.py (hunk 1 above)
Comment (lazarevevgeny) [reply model-optimizer/extensions/front/Pack.py:39]: I haven't seen any issue with Swish. Probably it is a sporadic issue. If it is reproducible on your machine I need to take a look.

**Comment by lazarevevgeny** [comment]
@aalborov, please, review the new document on how to convert the model.

**Comment by dkurt** [comment]
@lazarevevgeny, let's try to align OpenVINO class predictions with TensorFlow (for now `ov_class + 1 == tf_class`)

**Comment by lazarevevgeny** [comment]
> @lazarevevgeny, let's try to align OpenVINO class predictions with TensorFlow (for now `ov_class + 1 == tf_class`)

It can be easily done by inserting Add operation with constant [0, 1, 0, 0, 0, 0, 0] but then the object_detection_sample_ssd will be broken because it expects the output operation to be equal to DetectionOutput. I have described in the documentation the format of TF and IE models output so it will be easy to post-process the data.

**Review by dkurt — APPROVED** [review]

File: docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_EfficientDet_Models.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+Follow the instructions below to get frozen TensorFlow EfficientDet model. We use EfficientDet-D4 model as an example:
+
+1. Clone the repository:<br>
+```sh
+git clone https://github.com/google/automl
+cd automl/efficientdet
+```
+2. (Optional) Checkout to the commit that the conversion was tested on:<br>
+```sh
+git checkout 96e1fee
```
Comment (dkurt) [inline docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_EfficientDet_Models.md:22]: an error `AttributeError: module 'tensorflow_core.python.keras.api._v2.keras.initializers' has no attribute 'variance_scaling'` will be fixed by https://github.com/google/automl/pull/846

Comment (lazarevevgeny) [reply docs/MO_DG/prepare_model/convert_model/tf_specific/Convert_EfficientDet_Models.md:22]: Updated the documentation with a reference to this fix.

**Review by mvafin — APPROVED** [review]

**Pull request merged**